### PR TITLE
Divyanshu | Test | Support of US tax in Giddh 

### PR DESCRIPTION
**Pull request by dvCodeWorld**
https://app.clickup.com/t/86cvhwu2c

### CODE REVIEW

File: apps/web-giddh/src/app/vat-report/vat-report-filters/vat-report-filters.component.ts [hunk 1]
```diff
@@ -189,6 +227,105 @@ export class VatReportFiltersComponent implements OnInit {
...
+                        });
+                        this.taxAuthorityList = arr;
+                        if (this.taxAuthority.taxAuthorityUniqueName) {
+                            this.setTaxAuthorityLabel(this.taxAuthority.taxAuthorityUniqueName);
+                        }
+                    }
+                });
+            this.store.dispatch(this.settingsTaxesActions.GetTaxAuthorityList());
+
+            this.store.pipe(select(state => state.company && state.company.taxes), takeUntil(this.destroyed$)).subscribe(taxes => {
```
Comment (dilpreetsinghofficial) [inline apps/web-giddh/src/app/vat-report/vat-report-filters/vat-report-filters.component.ts:250]: Move to component store

File: apps/web-giddh/src/app/vat-report/vat-report-filters/vat-report-filters.component.ts [hunk 2]
```diff
@@ -189,6 +227,105 @@ export class VatReportFiltersComponent implements OnInit {
         });
         this.currentOrganizationType = this.generalService.currentOrganizationType;
         this.getCurrentCompanyBranches();
+
+        // Get All Tax Authorities for US Country
+        if (this.isUSCompany) {
+            this.store
+                .pipe(select(p => p.settings && p.settings.taxAuthorities), takeUntil(this.destroyed$))
```
Comment (dilpreetsinghofficial) [inline apps/web-giddh/src/app/vat-report/vat-report-filters/vat-report-filters.component.ts]: move to component store

File: apps/web-giddh/src/app/theme/tax-authority/utility/tax-authority.store.ts [hunk 3]
```diff
@@ -0,0 +1,301 @@
...
+    readonly exportSalesTaxReport = this.effect((data: Observable<{ reportType: string, params: any, isExport: boolean }>) => {
+        return data.pipe(
+            switchMap((req) => {
+                this.patchState({
+                    isLoading: true,
+                    exportTaxAuthorityWiseReport: null,
+                    exportTaxWiseReport: null,
+                    exportAccountWiseReport: null
+                });
+                return this.settingsTaxesService.GetExportSaleTaxReport(req.reportType, req.params, req.isExport).pipe(
```
Comment (dilpreetsinghofficial) [inline apps/web-giddh/src/app/theme/tax-authority/utility/tax-authority.store.ts]:
```suggestion
                return this.settingsTaxesService.getExportSaleTaxReport(req.reportType, req.params, req.isExport).pipe(
```

File: apps/web-giddh/src/app/theme/tax-authority/utility/tax-authority.store.ts [hunk 4]
```diff
@@ -0,0 +1,301 @@
...
+                            });
+                        }
+                    ),
+                    catchError((err) => EMPTY)
+                );
+            })
+        );
+    });
+
+    readonly exportSalesTaxReport = this.effect((data: Observable<{ reportType: string, params: any, isExport: boolean }>) => {
```
Comment (dilpreetsinghofficial) [inline apps/web-giddh/src/app/theme/tax-authority/utility/tax-authority.store.ts:292]: add documentation

File: apps/web-giddh/src/app/theme/tax-authority/utility/tax-authority.store.ts [hunk 5]
```diff
@@ -0,0 +1,301 @@
...
+    readonly getSalesTaxReport = this.effect((data: Observable<{ reportType: string, params: any, isExport: boolean }>) => {
+        return data.pipe(
+            switchMap((req) => {
+                this.patchState({
+                    isLoading: true,
+                    taxAuthorityWiseReport: null,
+                    taxWiseReport: null,
+                    accountWiseReport: null
+                });
+                return this.settingsTaxesService.GetExportSaleTaxReport(req.reportType, req.params, req.isExport).pipe(
```
Comment (dilpreetsinghofficial) [inline apps/web-giddh/src/app/theme/tax-authority/utility/tax-authority.store.ts]:
```suggestion
                return this.settingsTaxesService.getExportSaleTaxReport(req.reportType, req.params, req.isExport).pipe(
```

File: apps/web-giddh/src/app/theme/tax-authority/utility/tax-authority.store.ts [hunk 6]
```diff
@@ -0,0 +1,301 @@
...
+                            });
+                        }
+                    ),
+                    catchError((err) => EMPTY)
+                );
+            })
+        );
+    });
+
+    readonly getSalesTaxReport = this.effect((data: Observable<{ reportType: string, params: any, isExport: boolean }>) => {
```
Comment (dilpreetsinghofficial) [inline apps/web-giddh/src/app/theme/tax-authority/utility/tax-authority.store.ts:241]: Add documentation

File: apps/web-giddh/src/app/theme/tax-authority/utility/tax-authority.store.ts [hunk 7]
```diff
@@ -0,0 +1,301 @@
...
+                );
+            })
+        );
+    });
+
+    readonly deleteTaxAuthority = this.effect((data: Observable<string>) => {
+        return data.pipe(
+            switchMap((req) => {
+                this.patchState({ deleteTaxAuthorityIsSuccess: null });
+                return this.settingsTaxesService.DeleteTaxAuthority(req).pipe(
```
Comment (dilpreetsinghofficial) [inline apps/web-giddh/src/app/theme/tax-authority/utility/tax-authority.store.ts]:
```suggestion
                return this.settingsTaxesService.deleteTaxAuthority(req).pipe(
```

File: apps/web-giddh/src/app/theme/tax-authority/utility/tax-authority.store.ts [hunk 8]
```diff
@@ -0,0 +1,301 @@
...
+                            });
+                        }
+                    ),
+                    catchError((err) => EMPTY)
+                );
+            })
+        );
+    });
+
+    readonly deleteTaxAuthority = this.effect((data: Observable<string>) => {
```
Comment (dilpreetsinghofficial) [inline apps/web-giddh/src/app/theme/tax-authority/utility/tax-authority.store.ts:204]: add documentation

File: apps/web-giddh/src/app/theme/tax-authority/utility/tax-authority.store.ts [hunk 9]
```diff
@@ -0,0 +1,301 @@
...
+                );
+            })
+        );
+    });
+
+    readonly updateTaxAuthority = this.effect((data: Observable<any>) => {
+        return data.pipe(
+            switchMap((req) => {
+                this.patchState({ updateTaxAuthorityIsSuccess: null });
+                return this.settingsTaxesService.UpdateTaxAuthority(req.model, req.uniqueName).pipe(
```
Comment (dilpreetsinghofficial) [inline apps/web-giddh/src/app/theme/tax-authority/utility/tax-authority.store.ts]:
```suggestion
                return this.settingsTaxesService.updateTaxAuthority(req.model, req.uniqueName).pipe(
```

File: apps/web-giddh/src/app/theme/tax-authority/tax-authority.component.ts [hunk 10]
```diff
@@ -0,0 +1,133 @@
...
+    /**
+     * Lifecycle hook for destroy
+     *
+     * @memberof TaxAuthorityComponent
+     */
+    public ngOnDestroy(): void {
+        this.destroyed$.next(true);
+        this.destroyed$.complete();
+        document.querySelector('body').classList.remove('gst-sidebar-open');
+        this.asideGstSidebarMenuState === 'out'
```
Comment (dilpreetsinghofficial) [inline apps/web-giddh/src/app/theme/tax-authority/tax-authority.component.ts]:
```suggestion
        this.asideGstSidebarMenuState === 'out';
```

File: apps/web-giddh/src/app/theme/tax-authority/reports/tax-authority-report/tax-authority-report.component.ts [hunk 11]
```diff
@@ -0,0 +1,188 @@
...
+    /**
+    * Lifecycle hook for destroy
+    *
+    * @memberof TaxAuthorityReportComponent
+    */
+    public ngOnDestroy(): void {
+        this.destroyed$.next(true);
+        this.destroyed$.complete();
+        document.querySelector('body').classList.remove('gst-sidebar-open');
+        this.asideGstSidebarMenuState === 'out'
```
Comment (dilpreetsinghofficial) [inline apps/web-giddh/src/app/theme/tax-authority/reports/tax-authority-report/tax-authority-report.component.ts]:
```suggestion
        this.asideGstSidebarMenuState === 'out';
```

File: apps/web-giddh/src/app/theme/tax-authority/reports/tax-authority-report/tax-authority-report.component.ts [hunk 12]
```diff
@@ -0,0 +1,188 @@
...
+
+    /**
+     * Lifecycle hook for initialization
+     *
+     * @memberof TaxAuthorityReportComponent
+     */
+    public ngOnInit(): void {
+        document.querySelector('body').classList.add('gst-sidebar-open');
+        this.initSalesTaxReportForm();
+        this.store.pipe(select(state => state.session.activeCompany), takeUntil(this.destroyed$)).subscribe(activeCompany => {
```
Comment (dilpreetsinghofficial) [inline apps/web-giddh/src/app/theme/tax-authority/reports/tax-authority-report/tax-authority-report.component.ts]: move to component store

File: apps/web-giddh/src/app/theme/tax-authority/reports/tax-authority-report/tax-authority-report.component.html [hunk 13]
```diff
@@ -0,0 +1,157 @@
+<ng-container appTranslate [file]="'settings/tax-authority'" (localeData)="localeData = $event"
+    (commonLocaleData)="commonLocaleData = $event">
+    <hamburger-menu></hamburger-menu>
+    <!-- Tax Sidebar code  -->
+    <div class="relative w-100">
+        <div class="help-support-aside setting-sidebar-wrapper gst-sidebar-wrapper"
+            *ngIf="asideGstSidebarMenuState === 'in'">
+            <tax-sidebar (navigateEvent)="handleNavigation()"
+                (clickOutside)="isMobileScreen ? (asideGstSidebarMenuState = 'out') : ''"
+                [exclude]="'#primary-new-header'" class="gst-sidebar"></tax-sidebar>
```
Comment (dilpreetsinghofficial) [inline apps/web-giddh/src/app/theme/tax-authority/reports/tax-authority-report/tax-authority-report.component.html]: remove unused code

File: apps/web-giddh/src/app/theme/tax-authority/reports/rate-wise-report/rate-wise-report.component.ts [hunk 14]
```diff
@@ -0,0 +1,195 @@
...
+    /**
+    * Lifecycle hook for destroy
+    *
+    * @memberof RateWiseReportComponent
+    */
+    public ngOnDestroy(): void {
+        this.destroyed$.next(true);
+        this.destroyed$.complete();
+        document.querySelector('body').classList.remove('gst-sidebar-open');
+        this.asideGstSidebarMenuState === 'out'
```
Comment (dilpreetsinghofficial) [inline apps/web-giddh/src/app/theme/tax-authority/reports/rate-wise-report/rate-wise-report.component.ts]:
```suggestion
        this.asideGstSidebarMenuState === 'out';
```

File: apps/web-giddh/src/app/theme/tax-authority/reports/rate-wise-report/rate-wise-report.component.scss [hunk 15]
```diff
@@ -0,0 +1,19 @@
+.tax-authority-report {
+    max-width: var(--center-content-max-width);
+
+    .vat-report-table {
+        &.mat-mdc-table {
+            &.bordered-table {
+                tr {
```
Comment (dilpreetsinghofficial) [inline apps/web-giddh/src/app/theme/tax-authority/reports/rate-wise-report/rate-wise-report.component.scss]: move to global

File: apps/web-giddh/src/app/theme/tax-authority/reports/account-wise-report/account-wise-report.component.ts [hunk 16]
```diff
@@ -0,0 +1,199 @@
...
+    /**
+    * Lifecycle hook for destroy
+    *
+    * @memberof AccountWiseReportComponent
+    */
+    public ngOnDestroy(): void {
+        this.destroyed$.next(true);
+        this.destroyed$.complete();
+        document.querySelector('body').classList.remove('gst-sidebar-open');
+        this.asideGstSidebarMenuState === 'out'
```
Comment (dilpreetsinghofficial) [inline apps/web-giddh/src/app/theme/tax-authority/reports/account-wise-report/account-wise-report.component.ts]:
```suggestion
        this.asideGstSidebarMenuState === 'out';
```

File: apps/web-giddh/src/app/theme/tax-authority/reports/account-wise-report/account-wise-report.component.ts [hunk 17]
```diff
@@ -0,0 +1,199 @@
...
+    public ngOnInit(): void {
+        document.querySelector('body').classList.add('gst-sidebar-open');
+        this.initSalesTaxReportForm();
+        this.activateRoute.queryParams.pipe(takeUntil(this.destroyed$)).subscribe(queryParams => {
+            if (queryParams?.taxAuthorityUniqueName || queryParams?.taxUniqueName) {
+                this.getFormControl('taxAuthorityUniqueName').patchValue(queryParams?.taxAuthorityUniqueName ?? '');
+                this.getFormControl('taxUniqueName').patchValue(queryParams?.taxUniqueName ?? '');
+            }
+        });
+        this.store.pipe(select(state => state.session.activeCompany), takeUntil(this.destroyed$)).subscribe(activeCompany => {
```
Comment (dilpreetsinghofficial) [inline apps/web-giddh/src/app/theme/tax-authority/reports/account-wise-report/account-wise-report.component.ts]: move to component store

File: apps/web-giddh/src/app/theme/tax-authority/create/create.component.ts [hunk 18]
```diff
@@ -0,0 +1,144 @@
...
+        }
+    }
+
+    /**
+     * This will use for get states list
+     *
+     * @memberof CreateComponent
+     */
+    public getStates(): void {
+        this.store.pipe(select(state => state.general.states), takeUntil(this.destroyed$)).subscribe(response => {
```
Comment (dilpreetsinghofficial) [inline apps/web-giddh/src/app/theme/tax-authority/create/create.component.ts]: move to component store

File: apps/web-giddh/src/app/subscription/view-subscription/view-subscription.component.html [hunk 19]
```diff
@@ -75,6 +77,8 @@ <h4 class="font-16">{{ commonLocaleData?.app_view }} {{ localeData?.tab_heading
                                     activeCompany?.planVersion === 2
                                 "
                             >
+                                <!--   &&
+                                     activeCompany.subscription?.country?.countryCode !== 'GB' -->
```
Comment (dilpreetsinghofficial) [inline apps/web-giddh/src/app/subscription/view-subscription/view-subscription.component.html]: remove commented code

File: apps/web-giddh/src/app/subscription/subscription.component.html [hunk 20]
```diff
@@ -251,6 +251,8 @@
                                         mat-menu-item
                                         (click)="buyPlan(element)"
                                     >
+                                    <!-- &&
+                                            activeCompany.subscription?.country?.countryCode !== 'GB' -->
```
Comment (dilpreetsinghofficial) [inline apps/web-giddh/src/app/subscription/subscription.component.html]: remove commented code

File: apps/web-giddh/src/app/shared/aside-menu-create-tax/aside-menu-create-tax.component.ts [hunk 21]
```diff
@@ -212,12 +243,12 @@ export class AsideMenuCreateTaxComponent implements OnInit, OnChanges, OnDestroy
...
                 this.taxListSource$ = observableOf(this.taxList);
             } else {
-                this.store.dispatch(this._settingsTaxesActions.GetTaxList(countryCode));
+                this.store.dispatch(this.settingsTaxesActions.GetTaxList(countryCode));
             }
         });
     }
 
-    public selectTax(event) {
+    public selectTax() {
```
Comment (dilpreetsinghofficial) [inline apps/web-giddh/src/app/shared/aside-menu-create-tax/aside-menu-create-tax.component.ts]: Add return type and documentation

File: apps/web-giddh/src/app/shared/aside-menu-create-tax/aside-menu-create-tax.component.ts [hunk 22]
```diff
@@ -86,6 +98,19 @@ export class AsideMenuCreateTaxComponent implements OnInit, OnChanges, OnDestroy
                 }
             });
 
+        this.store
+            .pipe(select(p => p.settings && p.settings.taxAuthorities), takeUntil(this.destroyed$))
```
Comment (dilpreetsinghofficial) [inline apps/web-giddh/src/app/shared/aside-menu-create-tax/aside-menu-create-tax.component.ts]: move to component store

File: apps/web-giddh/src/app/shared/aside-menu-create-tax/aside-menu-create-tax.component.html [hunk 23]
```diff
@@ -12,7 +12,29 @@ <h2 class="aside-title" tabindex="0">{{ tax && tax?.uniqueName ? localeData?.upd
...
-                    <form #taxForm="ngForm" (ngSubmit)="onSubmit()">
+                    <div *ngIf="isTaxAuthoritiesLoading$ | async">
+                        <giddh-page-loader></giddh-page-loader>
+                    </div>
+                    <form #taxForm="ngForm" (ngSubmit)="onSubmit()" *ngIf="!(isTaxAuthoritiesLoading$ | async)">
+
+                        <div class="row" *ngIf="isUSCompany">
+                            <div class="form-group col-sm-6 col-xs-12">
+                                <label>{{ commonLocaleData?.app_tax_authority }} <span class="text-danger">*</span> </label
+                                    ><br />
```
Comment (dilpreetsinghofficial) [inline apps/web-giddh/src/app/shared/aside-menu-create-tax/aside-menu-create-tax.component.html:23]: confirm with @ravinderwalkover sir

File: apps/web-giddh/src/app/settings/taxes/setting.taxes.component.ts [hunk 24]
```diff
@@ -99,6 +100,14 @@ export class SettingTaxesComponent implements OnInit, OnDestroy {
                     this.toggleTaxAsidePane();
                 }
             });
+
+        this.store
+            .pipe(select(p => p.company && p.company.isTaxUpdatedSuccessfully), takeUntil(this.destroyed$))
```
Comment (dilpreetsinghofficial) [inline apps/web-giddh/src/app/settings/taxes/setting.taxes.component.ts]: move to component store

File: apps/web-giddh/src/app/settings/settings.component.ts [hunk 25]
```diff
@@ -249,7 +249,7 @@ export class SettingsComponent implements OnInit, OnDestroy {
             this.store.dispatch(this._generalActions.setAppTitle('/pages/settings/' + tab + '/' + this.integrationtab));
             this.loadModuleData(tab);
             this.router.navigate(['pages/settings/', tab, this.integrationtab], { replaceUrl: true });
-        } else {
+        } else if(tab !== 'reports') {
```
Comment (dilpreetsinghofficial) [inline apps/web-giddh/src/app/settings/settings.component.ts]: format this code

File: apps/web-giddh/src/app/settings/branch/create-branch/create-branch.component.ts [hunk 26]
```diff
@@ -250,11 +250,15 @@ export class CreateBranchComponent implements OnInit, OnDestroy {
         const requestObj = {
             name: formValue.name,
             alias: formValue.alias,
-            linkAddresses: formValue.address?.map(filteredAddress => ({
+            linkAddresses: []
+        };
+
+        if (formValue.address?.length) {
```
Comment (dilpreetsinghofficial) [inline apps/web-giddh/src/app/settings/branch/create-branch/create-branch.component.ts:256]:
```suggestion
        if (formValue?.address?.length) {
```

File: apps/web-giddh/src/app/services/settings.taxes.service.ts [hunk 27]
```diff
@@ -60,4 +62,109 @@ export class SettingsTaxesService {
...
+            case SalesTaxReport.AccountWise: apiUrl +=
+                (isExport ? COMPANY_API.EXPORT_SALES_ACCOUNT_WISE_REPORT : COMPANY_API.GET_SALES_TAX_ACCOUNT_WISE_REPORT)
+                ?.replace(':taxAuthorityUniqueName', encodeURIComponent(params?.taxAuthorityUniqueName))
+                ?.replace(':taxUniqueName', encodeURIComponent(params?.taxUniqueName))
+                break;
+        }
+        return this.http.get(apiUrl
+            ?.replace(':companyUniqueName', encodeURIComponent(this.companyUniqueName))
+            ?.replace(':taxNumber', encodeURIComponent(params?.taxNumber))
+            ?.replace(':count', encodeURIComponent(params?.count))
```
Comment (dilpreetsinghofficial) [inline apps/web-giddh/src/app/services/settings.taxes.service.ts:160]: check this code

**Review by dilpreetsinghofficial — CHANGES_REQUESTED** [review]

File: apps/web-giddh/src/app/vat-report/vat-report-filters/vat-report-filters.component.ts (hunk 1 above)
Comment (dvCodeWorld) [reply apps/web-giddh/src/app/vat-report/vat-report-filters/vat-report-filters.component.ts:249]: Done for Tax Authority

File: apps/web-giddh/src/app/vat-report/vat-report-filters/vat-report-filters.component.ts (hunk 2 above)
Comment (dvCodeWorld) [reply apps/web-giddh/src/app/vat-report/vat-report-filters/vat-report-filters.component.ts]: Done

File: apps/web-giddh/src/app/theme/tax-authority/utility/tax-authority.store.ts (hunk 3 above)
Comment (dvCodeWorld) [reply apps/web-giddh/src/app/theme/tax-authority/utility/tax-authority.store.ts]: Done for all

File: apps/web-giddh/src/app/theme/tax-authority/utility/tax-authority.store.ts (hunk 4 above)
Comment (dvCodeWorld) [reply apps/web-giddh/src/app/theme/tax-authority/utility/tax-authority.store.ts:292]: Done for all

File: apps/web-giddh/src/app/theme/tax-authority/tax-authority.component.ts (hunk 10 above)
Comment (dvCodeWorld) [reply apps/web-giddh/src/app/theme/tax-authority/tax-authority.component.ts]: Done

File: apps/web-giddh/src/app/theme/tax-authority/reports/tax-authority-report/tax-authority-report.component.ts (hunk 12 above)
Comment (dvCodeWorld) [reply apps/web-giddh/src/app/theme/tax-authority/reports/tax-authority-report/tax-authority-report.component.ts]: Done for all 3 reports

File: apps/web-giddh/src/app/theme/tax-authority/reports/tax-authority-report/tax-authority-report.component.html (hunk 13 above)
Comment (dvCodeWorld) [reply apps/web-giddh/src/app/theme/tax-authority/reports/tax-authority-report/tax-authority-report.component.html]: Done

File: apps/web-giddh/src/app/settings/branch/create-branch/create-branch.component.ts (hunk 26 above)
Comment (dvCodeWorld) [reply apps/web-giddh/src/app/settings/branch/create-branch/create-branch.component.ts:256]: no required

File: apps/web-giddh/src/app/theme/tax-authority/reports/rate-wise-report/rate-wise-report.component.scss (hunk 15 above)
Comment (dvCodeWorld) [reply apps/web-giddh/src/app/theme/tax-authority/reports/rate-wise-report/rate-wise-report.component.scss]: Done

File: apps/web-giddh/src/app/services/settings.taxes.service.ts (hunk 27 above)
Comment (dvCodeWorld) [reply apps/web-giddh/src/app/services/settings.taxes.service.ts:160]: Done

File: apps/web-giddh/src/app/settings/taxes/setting.taxes.component.ts (hunk 24 above)
Comment (dvCodeWorld) [reply apps/web-giddh/src/app/settings/taxes/setting.taxes.component.ts]: No Need to do.

File: apps/web-giddh/src/app/vat-report/vat-report-filters/vat-report-filters.component.ts (hunk 1 above)
Comment (dvCodeWorld) [reply apps/web-giddh/src/app/vat-report/vat-report-filters/vat-report-filters.component.ts:250]: No required

**Review by dilpreetsinghofficial — APPROVED** [review]

File: apps/web-giddh/src/app/add-company/add-company.component.html [hunk 28]
```diff
@@ -336,7 +336,7 @@
                                 </dropdown-field>
                             </div>
                             <ng-container
-                                *ngIf="secondStepForm.controls['businessType']?.value === businessTypes.Registered"
+                                *ngIf="secondStepForm.controls['businessType']?.value === businessTypes.Registered && this.selectedCountryCode !== 'US'"
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/add-company/add-company.component.html]: make a list of countries for which taxes are not applicable and check that list here

File: apps/web-giddh/src/app/settings/taxes/setting.taxes.component.html [hunk 29]
```diff
@@ -30,6 +30,7 @@ <h1 class="section-head m-0 pl-0 pb-0 pr-0">{{ localeData?.added_taxes }}</h1>
                                 <th>#</th>
                                 <th>{{ localeData?.tax }}</th>
                                 <th>{{ commonLocaleData?.app_name }}</th>
+                                <th *ngIf="availableTaxes[0]?.taxAuthority">{{ commonLocaleData?.app_tax_authority }}</th>
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/settings/taxes/setting.taxes.component.html]:
```suggestion
                                <th *ngIf="availableTaxes?.length && availableTaxes[0]?.taxAuthority">{{ commonLocaleData?.app_tax_authority }}</th>
```

File: apps/web-giddh/src/app/settings/taxes/setting.taxes.component.ts [hunk 30]
```diff
@@ -99,6 +99,14 @@ export class SettingTaxesComponent implements OnInit, OnDestroy {
...
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/settings/taxes/setting.taxes.component.ts]: don't use single character variable

File: apps/web-giddh/src/app/shared/tax-sidebar/tax-sidebar.component.html [hunk 31]
```diff
@@ -229,8 +229,60 @@
                             <span>{{ localeData?.filing_status }}</span>
                         </a>
                     </li>
+                    <li *ngIf="showSalesTaxMenus && isUSCompany">
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/shared/tax-sidebar/tax-sidebar.component.html]: add ng-container and add this if condition there and remove condition from all list elements

File: apps/web-giddh/src/app/theme/tax-authority/create/create.component.html [hunk 32]
```diff
@@ -0,0 +1,49 @@
...
+                        [name]="'name'"
+                        [label]="commonLocaleData?.app_description"
+                        [placeholder]="localeData?.enter_description"
+                        [type]="'text'"
+                        formControlName="description"
+                    >
+                    </input-field>
+                </div>
+                <div class="mr-t3">
+                    <button mat-stroked-button color="primary" [disabled]="taxAuthorityForm.invalid" (click)="createUpdateTaxAuthority()">
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/theme/tax-authority/create/create.component.html]: don't disable button and show error

File: apps/web-giddh/src/app/theme/tax-authority/tax-authority.component.ts [hunk 33]
```diff
@@ -0,0 +1,131 @@
...
+     * @memberof TaxAuthorityComponent
+     */
+    public openCreateUpdateTaxAuthorityDialog(isUpdateMode: boolean = false, taxAuthorityInfo?: any): void {
+        const dialogConfig: MatDialogConfig = {
+            width: 'var(--aside-pane-width)',
+            height: '100vh',
+            position: {
+                top: '0',
+                right: '0'
+            }
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/theme/tax-authority/tax-authority.component.ts]:
```suggestion
            },
            data: isUpdateMode ? taxAuthorityInfo : null
```

**Review by ravinderwalkover — CHANGES_REQUESTED** [review]

File: apps/web-giddh/src/app/add-company/add-company.component.html (hunk 28 above)
Comment (dvCodeWorld) [reply apps/web-giddh/src/app/add-company/add-company.component.html]: Done

File: apps/web-giddh/src/app/settings/taxes/setting.taxes.component.html (hunk 29 above)
Comment (dvCodeWorld) [reply apps/web-giddh/src/app/settings/taxes/setting.taxes.component.html]: Done

File: apps/web-giddh/src/app/settings/taxes/setting.taxes.component.ts (hunk 30 above)
Comment (dvCodeWorld) [reply apps/web-giddh/src/app/settings/taxes/setting.taxes.component.ts]: Done

File: apps/web-giddh/src/app/shared/tax-sidebar/tax-sidebar.component.html (hunk 31 above)
Comment (dvCodeWorld) [reply apps/web-giddh/src/app/shared/tax-sidebar/tax-sidebar.component.html]: Done

File: apps/web-giddh/src/app/theme/tax-authority/tax-authority.component.ts (hunk 33 above)
Comment (dvCodeWorld) [reply apps/web-giddh/src/app/theme/tax-authority/tax-authority.component.ts]: Done

File: apps/web-giddh/src/app/theme/tax-authority/create/create.component.html (hunk 32 above)
Comment (dvCodeWorld) [reply apps/web-giddh/src/app/theme/tax-authority/create/create.component.html]: Done

**Review by ravinderwalkover — APPROVED** [review]

**Pull request merged**